src/main/kotlin/Mempool.kt

```diff
@@ -0,0 +1,19 @@
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import java.net.URL
+
+
+class MempoolSpace {
+    private val json = Json { ignoreUnknownKeys = true }
+
+
+    suspend fun getBlocks(startHeight: Int): BlockResponse {
+        val url = URL("https://mempool.space/api/v1/blocks/$startHeight")
+        val blockJson = url.readText()
+        return json.decodeFromString(BlockResponse.serializer(), blockJson)
+    }
```

Comment:
As mentioned in my other comment, you should use ktor. It could for example look something like that:

```suggestion
    private val httpClient = HttpClient(CIO)

    suspend fun getBlocks(startHeight: Int): BlockResponse {
        val response = httpClient.get("https://mempool.space/api/v1/blocks/$startHeight")
        println(response.status)
        // See https://ktor.io/docs/serialization-client.html#serialization_dependency for help on how to deserialize the result
        TODO("decode response content and handle errors")
    }
```

But you should really dive into the ktor documentation and change that code to fit your needs.